### ISSUE TITLE
Repository and image cleanup on zarf package remove

### ISSUE THREAD

**Issue by blancharda** [defenseunicorns/zarf#1021]
### Overview
Currently `zarf package remove` only purges the target helm release, but does not remove the package's repos or images from the internal repository and registry. This behavior is somewhat unexpected, and could result in conflicts / confusion if a package is re-applied in the future.

### Suggested Change
Update the `zarf package remove` command to also remove package/component specific repos and images from the zarf repository and registry.

### Pros
* Aligns actual behavior with expected
* Reduces footprint when removing a package
* Avoids potential future conflicts when reapplying previously removed packages

### Cons / Challenges
* Will need to ensure repos/images are unique to the package/component being removed
* May cause unnecessary additional churn and data transfer in scenarios where package removal and redeployment are common

**Comment by RothAndrew** [comment]
(another data point, not speaking for the zarf team, just myself)

The behavior described is not undesirable to me. There have been several cases where I have been able to skip adding an image or repo because it was already added in another component or package.

If we were to do this, I'd prefer that it be done as an optional action, with the default being that the images/repos are not pruned.

A capability to easily list and prune/delete repos and/or images may be useful here, as a separate action from a `zarf package remove`

**Comment by blancharda** [comment]
Understood and agreed that it is sometimes desirable!

It's always nice when the artifact you're trying to push already happens to be there.
And similarly, we wouldn't want to delete an artifact upon the removal of PackageA, if that artifact was also in use by PackageB (see bullet #1 under Cons/Challenges).

But IMO it seems reasonable to expect that `zarf package remove` is the exact inverse of `zarf package deploy`, with the exception of NOT removing artifacts needed by remaining packages. Monotonic increase in storage space over time is never a good feature.

I like the idea of maintenance commands as well. Something like that seems like a necessity for long lived deployments.

**Comment by jeff-mccoy** [comment]
Yeah it's a bit more work, but I think we could also read all the packages and verify if an image or repo is used elsewhere too as a safety net.

**Comment by blancharda** [comment]
Is there any association between package/components and images stored in the Zarf state?
If not, maybe that would simplify the account keeping..

**Comment by Noxsios** [comment]
Partially implemented in: https://github.com/defenseunicorns/zarf/issues/1960

**Comment by brandtkeller** [comment]
Potentially destructive actions (like removing an artifact that may or may not be used elsewhere) is something we try to do very  carefully and with explicit definition. Adding an opt-in flag is reasonable and we have the image pruning as mentioned above. But we do not have any helpers for pruning the git server. 

Pruning during remove isolates the action to only the images (and potentially repos) that the package defines - which lowers the blast radius. 

We'll also need to ensure `prune` (or similar) across the commands makes sense and is intuitive
